modules/java/generator/src/cpp/common.h

```diff
@@ -16,6 +16,20 @@ extern "C" {
 #endif
 #include <jni.h>
 
+// make -fvisibility=hidden work with java 1.7
+#if defined(__linux__) && !defined (JNI_VERSION_1_8)
+  // adapted from jdk1.8/jni.h
+  #if (defined(__GNUC__) && ((__GNUC__ > 4) || (__GNUC__ == 4) && (__GNUC_MINOR__ > 2))) || __has_attribute(visibility)
+    #undef  JNIEXPORT
+    #define JNIEXPORT     __attribute__((visibility("default")))
+    #undef  JNIIMPORT
+    #define JNIIMPORT     __attribute__((visibility("default")))
+  #else
+    #define JNIEXPORT
```

Comment:
Perhaps we should not touch that case.

Hacks should be applied to well known cases only.

Reply:
I added an extra check for android, so I have a patch for ourselves at least.